### ISSUE TITLE
fix xpath in charters:years()

### ISSUE THREAD

**Issue by lindesc** [icaruseu/mom-ca#16]

**Comment by strec** [comment]
In which context is the current charters:years() making problems? I saw that it complains:
"Error SENR0001: attribute 'from' has no parent element" for different reasons (node vs. atomic values)
```
for $year in $charter-base-collection//cei:text//(cei:date/@value|cei:dateRange/@from)
    order by xs:integer($year)
    return
    $year
```
This error can be circumvented by using
`
for $year in $charter-base-collection//cei:text//(cei:date/string(@value)|cei:dateRange/string(@from))
` 
for example, but then the union between the two node-sets won't work, because they are two string-sets now.

A construct like the following does the trick:
```
for $year in ($charter-base-collection//cei:text//cei:date/string(@value), $charter-base-collection//cei:text//cei:dateRange/string(@from))
```
but then you get different errors, from different parts of the application framework, that seem to depend on the function charters:years() returning the type attribute.

So what is wrong with the XPath as it is? I did not come across any problems in the application behaviour...

**Comment by strec** [comment]
Of course, the performance seems to be improvable, yet. Only in the rarest of cases are the cei:date and cei:dateRange not inside of cei:text elements. 

collection("/db/mom-data")//cei:date -> 268462 hits
collection("/db/mom-data")//cei:text//cei:date -> 268432 hits

collection("/db/mom-data")//cei:text//cei:dateRange -> 308447 hits
collection("/db/mom-data")//cei:dateRange -> 308452 hits

collection("/db/mom-data")//cei:text -> 579365 hits
collection("/db/mom-data")//cei:text[descendant::cei:date] -> 268413 hits
collection("/db/mom-data")//cei:text[descendant::cei:dateRange] -> 308444 hits
